src/App.tsx

```diff
@@ -1,223 +1,235 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
+import {
+  addTodo,
+  deleteTodo,
+  updateTodo,
+  getTodos,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ErrorMessage } from './components/ErrorMessage';
+import { FilterForTodos } from './components/FilterForTodos';
+import { NewTodo } from './components/NewTodo';
+import { TodosList } from './components/TodosList';
+import { Todo } from './types/Todo';
+import { TodoStatus } from './types/TodoStatus';
+
+const emptyTodo: Todo = {
+  id: 0,
+  title: 'adding ...',
+  userId: 0,
+  completed: false,
+};
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<TodoStatus>(TodoStatus.ALL);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorText, setErrorText] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo>(emptyTodo);
+  const [isTodoAdding, setIsTodoAdding] = useState(false);
+  const [loadingTodosIds, setLoadingTodosIds] = useState<number[]>([0]);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const loadTodos = useCallback(async () => {
+    try {
+      if (user) {
+        const responseTodos = await getTodos(user.id);
+
+        setTodos(responseTodos);
+      }
+    } catch (error) {
+      setIsError(true);
+      setErrorText('Unable to load todos');
+
+      throw new Error(`unexpected error with loading todos: ${error}`);
```

Comment:
you're already handle the error via error and error text variables, throwing error here is redundant (you also didn't cover loadTodos call in try catch functionality)